packages/api/lib/writeRecords/write-granules.js

```diff
@@ -841,13 +856,29 @@ const writeGranulesFromMessage = async ({
   const executionDescription = await granuleModel.describeGranuleExecution(executionArn);
   const processingTimeInfo = getExecutionProcessingTimeInfo(executionDescription);
   const provider = getMessageProvider(cumulusMessage);
-  const workflowStartTime = getMessageWorkflowStartTime(cumulusMessage);
   const error = parseException(cumulusMessage.exception);
   const workflowStatus = getMetaStatus(cumulusMessage);
   const collectionId = getCollectionIdFromMessage(cumulusMessage);
   const pdrName = getMessagePdrName(cumulusMessage);
   const queryFields = getGranuleQueryFields(cumulusMessage);
 
+  let workflowStartTime;
+  try {
+    workflowStartTime = getMessageWorkflowStartTime(cumulusMessage);
+    if (!workflowStartTime) {
+      throw new Error(
```

Comment:
nit: do we want to throw the same error type `CumulusMessageError`?

Reply:
Not a nit at all, good catch: 5042938b30